Modix.Services/Starboard/StarboardService.cs

```diff
@@ -34,31 +89,41 @@ public async Task<IUserMessage> GetFromStarboard(IGuildChannel channel, IMessage
                 .GetMessagesAsync()
                 .FlattenAsync();
 
+            //We need to store entries in the db to avoid this horrible mess
             return starMessages
                 .Cast<IUserMessage>()
-                .FirstOrDefault(x => x.Content.EndsWith(message.Id.ToString()));
+                .FirstOrDefault(x => x.Embeds
+                                        .Select(y => y.Fields.Last())
+                                        .Any(y => y.Value.Contains(message.GetJumpUrl())));
```

Comment:
Letting it be known that I hate the approach of parsing through the starboard messages hoping to find the right one. My vote goes to using the database.

Reply:
Not to mention that with this current implementation we're only looking through the 100 most recent messages. If the starboard-entry comes before that, we'll add it again, ergo, duplicate entries. I don't like this either. 

Now, with this said, we COULD download even more messages, and I shouldn't have to explain why this is bad should I?

Reply:
I would agree with Scott. It's not about downloading more messages (if by that you mean getting them from Discord), your query will be faster imo.